src/lib/mina_net2/keypair.ml

```diff
@@ -0,0 +1,65 @@
+open Core
+open Async
+open Network_peer
+
+[%%versioned
+module Stable = struct
+  module V1 = struct
+    type t = { secret : string; public : string; peer_id : Peer.Id.Stable.V1.t }
+
+    let to_latest = Fn.id
+  end
+end]
+
+let secret { secret; _ } = secret
+
+let generate_random helper =
+  match%map
+    Libp2p_helper.do_rpc helper
+      (module Libp2p_ipc.Rpcs.GenerateKeypair)
+      (Libp2p_ipc.Rpcs.GenerateKeypair.create_request ())
+  with
+  | Ok response ->
+      let open Libp2p_ipc.Reader in
+      let keypair =
+        Libp2pHelperInterface.GenerateKeypair.Response.result_get response
+      in
+      let peer_id =
+        keypair |> Libp2pKeypair.peer_id_get |> PeerId.id_get
+        |> Peer.Id.unsafe_of_string
+      in
+      let secret = Libp2pKeypair.private_key_get keypair in
```

Comment:
There is one issue with `ocaml-capnp` library: in case of broken encoding it will simply return an empty string here.

So does it for all of the other types (i.e. zero values on numeric types).

This leads to us accepting the data when we should have reset the stream.